### PR TITLE
Fix typo in JavaCompile.java

### PR DESCRIPTION
**Pull request by nugmanoff**
Fixes #?

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by ljacomet** [comment]
@nugmanoff Please resolve the DCO error otherwise we cannot accept the PR.

**Comment by nugmanoff** [comment]
@ljacomet resolved✅

File: subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java [hunk 1]
```diff
@@ -156,7 +156,7 @@ protected void compile(InputChanges inputs) {
     private void validateConfiguration() {
         if (javaCompiler.isPresent()) {
             checkState(getOptions().getForkOptions().getJavaHome() == null, "Must not use `javaHome` property on `ForkOptions` together with `javaCompiler` property");
-            checkState(getOptions().getForkOptions().getExecutable() == null, "Must not use `exectuable` property on `ForkOptions` together with `javaCompiler` property");
+            checkState(getOptions().getForkOptions().getExecutable() == null, "Must not use `executable` property on `ForkOptions` together with `javaCompiler` property");
```
Comment (big-guy) [inline subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/JavaCompile.java:159]: Can you also take a look at this test too? https://github.com/gradle/gradle/blob/master/subprojects/language-java/src/test/groovy/org/gradle/api/tasks/compile/JavaCompileTest.groovy#L50-L61

**Comment by nugmanoff** [comment]
@big-guy ✅

**Review by ljacomet — APPROVED** [review]

**Comment by ljacomet** [comment]
@bot-gradle test QF

**Comment by ljacomet** [comment]
@nugmanoff Could you rebase on the latest `master`? Some changes were done to the CI setup that need to be included to have this fix go through CI.

**Comment by nugmanoff** [comment]
@ljacomet done

**Comment by ljacomet** [comment]
@bot-gradle test QF

**Pull request merged**